tensorly/decomposition/_parafac2.py

```diff
@@ -340,35 +348,78 @@ def parafac_updates(X, w, f):
     for iteration in range(n_iter_max):
         if verbose:
             print("Starting iteration", iteration)
+
+        # Will we be performing a line search iteration?
+        if linesearch and iteration % 2 == 0 and iteration > 5:
+            line_iter = True
+            factors_last = [tl.copy(f) for f in factors]
+            factors_last[1] = factors_last[1] * T.reshape(weights, (1, -1))
+        else:
+            line_iter = False
+
         factors[1] = factors[1] * T.reshape(weights, (1, -1))
         weights = T.ones(weights.shape, **tl.context(tensor_slices[0]))
```

Comment:
```suggestion
        factors[1] = factors[1] * T.reshape(weights, (1, -1))
        weights = T.ones(weights.shape, **tl.context(tensor_slices[0]))
        
        # Will we be performing a line search iteration?
        if linesearch and iteration % 2 == 0 and iteration > 5:
            line_iter = True
            factors_last = [tl.copy(f) for f in factors]
        else:
            line_iter = False
```

Could we change here the order to avoid multiplying in the weights twice?